### PR TITLE
Dynamax messages

### LINKED ISSUE #2702: Add a trainer slide-in msg for dynamax/gigantamax when #2417 is merged
In Pokémon SwSh, trainer NPCs can have a battle dialogue when they're about to activate dynamax/gigantamax.

That cannot be properly tested right now even if we tried to implement it, because we don't have dynamax/gigantamax implemented yet.

Thus, I'm leaving this Issue ticket as a reminder until the day in which #2417 or a PR that supersedes it is merged.

### PR DESCRIPTION
**Pull request by Bassoonian**
Mega Evolution and Z-moves already had battle messages integrated. With Dynamax being a thing now as of #2417, the missing message documented in #2702 can now be fixed. Also changes the various for the Mega Evolution and Z-Move messages into callnatives.

## Issue(s) that this PR fixes
Fixes #2702 

## **Discord contact info**
bassoonian

### CODE REVIEW

**Comment by AlexOn1ine** [comment]
Can you change the various to a callnative? If you want you could also change the mega and z-moves ones to callnatives.

**Comment by Bassoonian** [comment]
Ready for re-review.

File: asm/macros/battle_script.inc [hunk 1]
```diff
@@ -1518,6 +1518,21 @@
 	.4byte \ptr
 	.endm
 
+	.macro trytrainerslidezmovemsg, battler:req
+	callnative BS_TryTrainerSlideZMoveMsg
+	.byte \battler
```
Comment (AlexOn1ine) [inline asm/macros/battle_script.inc:1523]: The battler argument isn't needed. You can just use `gBattlerAttacker` inside the callnative. E.g `BS_SetGlaiveRush`

**Comment by Bassoonian** [comment]
Ready for re-review!

**Review by AlexOn1ine — APPROVED** [review]

**Pull request merged**